# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/linux/linux26/74-asm-byteorder_h.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./include/linux/byteorder/little_endian.h.orig	2004-11-21 18:54:27.344956208 +0100
+++ ./include/linux/byteorder/little_endian.h	2004-11-21 18:54:36.247602800 +0100
@@ -8,6 +8,21 @@
 #define __LITTLE_ENDIAN_BITFIELD
 #endif
 
+#ifndef __KERNEL__
+/* fix for inclusion from userspace... :( */
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+#endif
+
 #include <linux/types.h>
 #include <linux/byteorder/swab.h>
 
--- ./include/linux/byteorder/big_endian.h.orig	2004-11-24 15:49:41.099573896 +0100
+++ ./include/linux/byteorder/big_endian.h	2004-11-24 15:50:46.048700128 +0100
@@ -8,6 +8,21 @@
 #define __BIG_ENDIAN_BITFIELD
 #endif
 
+#ifndef __KERNEL__
+/* fix for inclusion from userspace... :( */
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+#endif
+
 #include <linux/types.h>
 #include <linux/byteorder/swab.h>
 
